app/models/ems_refresh/save_inventory_cloud.rb

```diff
@@ -253,4 +255,31 @@ def save_resource_groups_inventory(ems, hashes, target = nil)
     save_inventory_multi(ems.resource_groups, hashes, deletes, [:ems_ref])
     store_ids_for_new_records(ems.resource_groups, hashes, :ems_ref)
   end
+
+  def save_cloud_services_inventory(ems, hashes, target = nil)
+    target = ems if target.nil?
+
+    ems.cloud_services.reset
+    deletes = if (target == ems)
+                :use_association
+              else
+                []
+              end
+
+    infra_ems = ems.try(:provider) && ems.provider.try(:infra_ems)
```

Comment:
please move L269-L279 to a refresh code, it's better to keep saving code as clean as possible, since it shared by all providers